psmdb/percona-server-for-mongodb-4.0-template.yml

```diff
@@ -39,314 +44,218 @@
-            exit 1
-          fi
-          tar xf percona-server-mongodb-${PSM_VERSION}-${PSM_RELEASE}.tar.gz
-          tar xf percona-server-mongodb-${PSM_VERSION}-${PSM_RELEASE}-${DISTRIBUTION}-x86_64.tar.gz
-          mv percona-server-mongodb-${PSM_VERSION}-${PSM_RELEASE}/bin/* percona-server-mongodb-${PSM_VERSION}-${PSM_RELEASE}
-          mv percona-server-mongodb-${PSM_VERSION}-${PSM_RELEASE} ${PSMDIR}
+        if [ ${BUILD_TYPE} = "debug" ]; then
+            TARBALL='debug'
+        else
+            TARBALL='tarball'
         fi
 
+        SOURSE_TAR_MDB=$(basename $(find $WORKSPACE/test/source_tarball -name 'percona-server-mongodb*.tar.gz' | sort | tail -n1))
+        cd $WORKSPACE/test/source_tarball/
+        tar -xzf $SOURSE_TAR_MDB
+        rm -rf $SOURSE_TAR_MDB
+
+
+        SOURS_DIR=$(ls)
+        cd $(ls)
+        mv ./jstests ../
+        mv ./buildscripts ../
+
+        mv $WORKSPACE/test/$SOURS_DIR/build ../
+        mv $WORKSPACE/test/$SOURS_DIR/src  ../
+
+        cd ${WORKSPACE}
+        TAR_MDB=$(basename $(find $WORKSPACE/test/$TARBALL -name 'percona-server-mongodb*.tar.gz' | sort | tail -n1))
+        cd $WORKSPACE/test/$TARBALL
+        tar -xzf $TAR_MDB
+        rm -rf $TAR_MDB
+
+        MONGO_DB_DIR=$(ls)
+        mv ./$MONGO_DB_DIR ./percona-server-mongodb
+        cd ./percona-server-mongodb
+        mv $WORKSPACE/test/source_tarball/jstests ./bin/
+        mv $WORKSPACE/test/source_tarball/buildscripts ./bin/
+        mv $WORKSPACE/test/source_tarball/build ./bin/
+        mv $WORKSPACE/test/source_tarball/src ./bin/
+
+        PSMDIR=$(basename $(pwd))
+        PSMDIR_ABS=$PWD
+
+        DISTRIBUTION="xenial"
+
+
         if [ "$JOBS" != "auto" ]; then
-          sed -i "s/^RESMOKE_JOBS=.*$/RESMOKE_JOBS=${JOBS}/" psmdb-misc-scripts/run_resmoke_psmdb_3.2.sh
+            sed -i "s/^RESMOKE_JOBS=.*$/RESMOKE_JOBS=${JOBS}/" $WORKSPACE/psmdb-misc-scripts/run_resmoke_psmdb_3.2.sh
         fi
+
         if [ -n "${CUSTOM_SUITE}" ]; then
-          echo "${CUSTOM_SUITE}" > psmdb-misc-scripts/suite_sets/custom.txt
-          TEST_SUITE_FILE="custom.txt"
+            echo "${CUSTOM_SUITE}" > $WORKSPACE/psmdb-misc-scripts/suite_sets/custom.txt
+            TEST_SUITE_FILE="custom.txt"
         else
-          TEST_SUITE_FILE="${SUITE}.txt"
+            TEST_SUITE_FILE="${SUITE}.txt"
         fi
 
-        cd ${PSMDIR_ABS}
+        cd ${PSMDIR_ABS}/bin/
         find build/opt -type f -name '*.a' -delete || true
         find build/opt -type f -name '*.o' -delete || true
         touch killer.log
+
         if [ "${ENABLE_KILLER}" = "true" ]; then
-          ../psmdb-misc-scripts/killer.sh &
+            $WORKSPACE/psmdb-misc-scripts/killer.sh &
         fi
 
-        for trial in `seq 1 ${NUM_TRIALS}`;
-        do
-        ../psmdb-misc-scripts/run_resmoke_psmdb_3.2.sh ${trial} ${TEST_SUITE_FILE}
+
+        for trial in `seq 1 ${NUM_TRIALS}`; do
+            ulimit -u 64000
```

Comment:
Why this here and in a loop?

Reply:
Thank you, it was for test. I have removed it.